udata/frontend/markdown.py

```diff
@@ -1,18 +1,52 @@
 # -*- coding: utf-8 -*-
 from __future__ import unicode_literals
 
+from urlparse import urlparse
+
 import bleach
 import CommonMark
 from html5lib.sanitizer import HTMLSanitizer
 from flask import current_app, Markup
 from werkzeug.local import LocalProxy
 from jinja2.filters import do_truncate, do_striptags
 
+from udata.i18n import _
+
 md = LocalProxy(lambda: current_app.extensions['markdown'])
 
 EXCERPT_TOKEN = '<!--- --- -->'
 
 
+def avoid_mailto_callback(attrs, new=False):
+    """Remove completely the link containing a `mailto`."""
+    if attrs['href'].startswith('mailto:'):
+        return None
+    return attrs
+
+
+def nofollow_callback(attrs, new=False):
+    """
+    Turn relative links into external ones and avoid `nofollow` for us,
+
+    otherwise add `nofollow`.
+    That callback is not splitted in order to parse the URL only once.
+    """
+    parsed_url = urlparse(attrs['href'])
+    if parsed_url.netloc in ('', current_app.config['SERVER_NAME']):
+        attrs['href'] = '{scheme}://{netloc}{path}'.format(
+            scheme=current_app.config['USE_SSL'] and 'https' or 'http',
+            netloc=current_app.config['SERVER_NAME'],
+            path=parsed_url.path)
+        attrs['data-tooltip'] = _('Source')  # For embeds.
```

Comment:
It's only for our internal links but I agree that's it's far from being perfect… suggestions? Callbacks don't have any extra parameter :/